features/FxA-41-signin-funnel-metrics/README.md

```diff
@@ -0,0 +1,305 @@
+# Sign-in funnel metrics
+
+## Problem Summary
+
+We know that
+a significant number of our users
+encounter problems when signing in,
+but we don't have a way to measure
+the causes or impact of those problems.
+For example, we know that
+many users reset their password when signing in,
+but we don't know how many eventually succeed
+and how many abandon the entire process.
+
+To help us improve
+the experience of signing in to FxA
+in a data-driven way,
+we need to analyze metrics
+about the entire sign-in flow
+as an end-to-end experience,
+rather than looking at individual events in isolation.
+
+## Outcomes
+
+We want a more accurate understanding
+of how real users experience
+the sign-in and sign-up flows.
+Concretely, this will be delivered
+by a set of graphs that render the aggregated flows
+as a user-conversion funnel.
+Analyzing these graphs will enable us
+to better decide
+where we should focus future development efforts.
+
+This feature will be successful if
+we can confidently identify
+the two most common reasons
+why users fail to connect to Sync
+and propose appropriate changes
+to reduce occurrences of them.
+
+## Hypothesis
+
+We believe that
+rendering sign-in metrics as a user-conversion funnel
+will help us make clearer decisions
+on how to increase the success-rate
+of users connecting to Sync.
+
+We will know this to be true
+when product-level conversations about features
+are driven by the information conveyed in these graphs.
+
+## Metrics
+
+The only metrics for this feature
+are the binary questions:
+
+* Do the required graphs exist?
+* Do they provide the expected value?
+
+## Detailed design
+
+We've found that
+the open-ended, exploratory nature of redash
+can encourage discovery of unexpected traits and trends in data.
+We'll utilize that benefit for this feature
+by emitting new "flow events"
+from the content and auth servers,
+which can be imported to redshift
+in a similar fashion to our activity events.
+Having data for both types of event
+side-by-side in the same database
+may also support other analysis
+not considered here.
+
+The following flow events
+represent discrete stages
+in a successful flow:
+
+|Name|Origin|Description|
+|----|------|-----------|
+|`flow.begin`|content|User has landed on a page that allows them to log in or sign up.|
+|`login.engage`|content|User has interacted with the login form at `/signin` or `/force_auth`.|
+|`signup.engage`|content|User has interacted with the signup form at `/signup`.|
```

Comment:
It might over-count `signup.engage` vs `account.created` and make the drop-off at the stage look bigger than it should.  But we can cross that bridge when we come to it.